qds_sdk/clusterv2.py

```diff
@@ -539,7 +566,25 @@ def clone(cls, cluster_id_label, cluster_info):
         conn = Qubole.agent(version="v2")
         return conn.post(cls.element_path(cluster_id_label) + '/clone', data=cluster_info)
 
-    # implementation needed
     @classmethod
-    def list(self, state=None):
-        pass
\ No newline at end of file
+    def list(cls, state=None):
+        """
+        List existing clusters present in your account.
+
+        Kwargs:
+            `state`: list only those clusters which are in this state
+
+        Returns:
+            List of clusters satisfying the given criteria
+        """
+        conn = Qubole.agent(version="v2")
+        if state is None:
+            return conn.get(cls.rest_entity_path)
+        elif state is not None:
+            cluster_list = conn.get(cls.rest_entity_path)
+            result = []
+            if 'clusters' in cluster_list:
+                for cluster in cluster_list['clusters']:
+                    if state.lower() == cluster['state'].lower():
+                        result.append(cluster)
+            return result
```

Comment:
Can rewrite this fn to something like:
```
conn = Qubole.agent(version="v2")
cluster_list = conn.get(cls.rest_entity_path)
if state is None: 
   # return the complete list since state is None
   return cluster_list
# filter clusters based on state
result = []
if 'clusters' in cluster_list:
   for cluster in cluster_list['clusters']:
        if state.lower() == cluster['state'].lower():
             result.append(cluster)
return result
```